src/Eloquent/Builder.php

```diff
@@ -195,6 +196,10 @@ public function raw($value = null)
      */
     public function createOrFirst(array $attributes = [], array $values = []): Model
     {
+        if ($attributes === []) {
+            throw new InvalidArgumentException('You must provide attributes to check for duplicates');
+        }
```

Comment:
I'd suggest adding a comment about conflicts between `$attributes` and `$values`, regardless of whether that issue gets reported upstream in Eloquent.

Re: https://github.com/mongodb/laravel-mongodb/pull/2742#discussion_r1520418945